src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -312,21 +336,26 @@ internal void Configure()
                 DetermineIgnoreCondition();
                 DetermineSerializationCapabilities();
             }
+        }
 
-            if (IsRequired)
+        internal void CheckRequiredPropertyIsMisused()
```

Comment:
Nit: "Misused" sounds ambiguous. Recommend renaming to `ValidateRequiredPropertyConfiguration` or something along those lines. Assuming the check gets moved to the `Configure()` method it should also be marked private.

Reply:
fair, will change - I don't have strong preference on naming